awscli/autocomplete/autogen.py

```diff
@@ -0,0 +1,177 @@
+"""This module is used to auto-generate server side completion data.
+
+This module can be used during the build process to generate any new
+server side autocompletions based on updated service models.
+It can also be used to regen completion data as new heuristics are added.
+
+"""
+import logging
+from difflib import SequenceMatcher
+from collections import namedtuple
+
+
+LOG = logging.getLogger(__name__)
+Resource = namedtuple('Resource', ['resource_name', 'ident_name',
+                                   'operation', 'jp_expr'])
+
+
+class ServerCompletionHeuristic(object):
+    # Prefix that typically indicates listing resources,
+    # e.g ListResources, DescribeResources, etc.
+    _RESOURCE_VERB_PREFIX = ('list', 'describe')
```

Comment:
We may need to add `get`, but not sure how that affects the results. I know some services such as API Gateway use `get` instead of `list` and `describe`

Reply:
I went ahead and added it.  It's not great but it's better than nothing.  Here's an example of api gateway: https://gist.github.com/jamesls/043bb24bd2a6efd427109f72be2d7eca

Reply:
Actually, I'll have to remove this for now.  Right now we make the assumption that the operations fall into two buckets, one for auto completing server side values, and the other for operations that *can* be auto-completed.  `Get` can be used for both cases so until there's more sophisticated logic here it can't be part of `_RESOURCE_VERB_PREFIX`.